app/lib/service/openid/jwks.dart

```diff
@@ -0,0 +1,97 @@
+// Copyright (c) 2022, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+import 'dart:convert';
+import 'dart:typed_data';
+
+import 'package:json_annotation/json_annotation.dart';
+
+part 'jwks.g.dart';
+
+/// The list of JSON Web Key records.
+@JsonSerializable(includeIfNull: false, explicitToJson: true)
+class JsonWebKeyList {
+  final List<JsonWebKey> keys;
+
+  JsonWebKeyList({
+    required this.keys,
+  });
+
+  factory JsonWebKeyList.fromJson(Map<String, dynamic> json) =>
+      _$JsonWebKeyListFromJson(json);
+
+  Map<String, dynamic> toJson() => _$JsonWebKeyListToJson(this);
+}
+
+/// The JSON Web Key record.
+@JsonSerializable(includeIfNull: false, explicitToJson: true)
```

Comment:
Need to reference the specification:
https://datatracker.ietf.org/doc/html/rfc7517
And list of registered parameters:
https://www.iana.org/assignments/jose/jose.xhtml#web-key-parameters

I would suggest supporting all possible parameters.


See also: https://github.com/google/webcrypto.dart/blob/master/lib/src/jsonwebkey.dart
We shouldn't use that, it's intended for communication with webcrypto in the browser :D